src/other/index.jl

```diff
@@ -174,9 +174,14 @@ end
     idx
 end
 
-@inline Base.getindex(x::AbstractIndex, idx::AbstractRange{<:Integer}) = getindex(x, collect(Int, idx))
+@inline Base.getindex(x::AbstractIndex, idx::AbstractRange{<:Integer}) =
+    getindex(x, collect(Int, idx))
 @inline Base.getindex(x::AbstractIndex, ::Colon) = Base.OneTo(length(x))
-@inline Base.getindex(x::AbstractIndex, notidx::Not) = setdiff(1:length(x), getindex(x, notidx.skip))
+@inline Base.getindex(x::AbstractIndex, notidx::Not) =
+    setdiff(1:length(x), getindex(x, notidx.skip))
+@inline Base.getindex(x::AbstractIndex, idx::Between) = collect(x[idx.first]:x[idx.last])
+@inline Base.getindex(x::AbstractIndex, idx::All) =
+    idx.cols == () ? collect(1:length(x)) : union(getindex.(Ref(x), idx.cols)...)
```

Comment:
```suggestion
    isempty(idx.cols) ? collect(1:length(x)) : union(getindex.(Ref(x), idx.cols)...)
```
(To avoid recompiling `union` for each number of arguments.)